Sources/Core/Sources/Render/World/WorldRenderer.swift

```diff
@@ -37,6 +37,15 @@ public final class WorldRenderer: Renderer {
   private let blockOutlineVertexBuffer: MTLBuffer
   /// A buffer containing indices for a block outline.
   private let blockOutlineIndexBuffer: MTLBuffer
+  
+  private var renderBlockOutline: Bool {
+    let playerGamemode = client.game.currentGamemode()
+    guard let playerGamemode else {
```

Comment:
I think this check is unnecessary, the direct comparison with `.spectator` should work for an optional too

Reply:
Do you mean... the entire computed variable, doing `client.game.currentGamemode()` in `render()` or just getting rid of guard?